eng/pipelines/handlers.yml

```diff
@@ -204,55 +204,55 @@ stages:
-                displayName: 'Install .NET (Local Workloads)'
-                retryCountOnTaskFailure: 3
-                env:
-                  DOTNET_TOKEN: $(dotnetbuilds-internal-container-read-token)
-                  PRIVATE_BUILD: $(PrivateBuild)
-              - pwsh: ./build.ps1 --target=dotnet-samples --configuration="${{ BuildConfiguration }}" --verbosity=diagnostic
-                displayName: 'Build .NET MAUI Samples'
-              - task: PublishBuildArtifacts@1
-                condition: always()
-                displayName: publish artifacts
-                inputs:
-                  ArtifactName: ${{ BuildPlatform.artifact }}
+        - job: build_net_${{ BuildPlatform.name }}_samples
+          workspace:
+            clean: all
+          displayName: ${{ BuildPlatform.name }}
+          timeoutInMinutes: 120
+          pool:
+            name: ${{ BuildPlatform.poolName }}
+            vmImage: ${{ BuildPlatform.vmImage }}
+            demands:
+              - macOS.Name -equals Monterey
+              - macOS.Architecture -equals x64
+              - Agent.HasDevices -equals False
+              - Agent.IsPaired -equals False
+          steps:
+            - template: common/provision.yml
+              parameters:
+                platform: ${{ BuildPlatform.name }}
+                poolName: ${{ BuildPlatform.poolName }}
+
+            - task: DownloadBuildArtifacts@0
+              displayName: 'Download Packages'
+              inputs:
+                artifactName: nuget
+                itemPattern: '**/*.nupkg'
+                downloadPath: $(System.DefaultWorkingDirectory)/artifacts
+            - pwsh: Move-Item -Path artifacts\nuget\*.nupkg -Destination artifacts -Force
+              displayName: Move the downloaded artifacts
+            - pwsh: ./build.ps1 --target=dotnet-local-workloads --verbosity=diagnostic
+              displayName: 'Install .NET (Local Workloads)'
+              retryCountOnTaskFailure: 3
+              env:
+                DOTNET_TOKEN: $(dotnetbuilds-internal-container-read-token)
+                PRIVATE_BUILD: $(PrivateBuild)
+            - task: DotNetCoreCLI@2
+              inputs:
+                projects: $(Build.SourcesDirectory)/src/TestUtils/src/Microsoft.Maui.IntegrationTests/Microsoft.Maui.IntegrationTests.csproj
+              displayName: Build Microsoft.Maui.IntegrationTests
+            - task: DotNetCoreCLI@2
+              inputs:
+                command: test
+                projects: $(Build.SourcesDirectory)/src/TestUtils/src/Microsoft.Maui.IntegrationTests/bin/Debug/net7.0/Microsoft.Maui.IntegrationTests.dll
+                arguments: --logger "console;verbosity=normal" --filter "FullyQualifiedName=Microsoft.Maui.IntegrationTests.SampleTests"
+                publishTestResults: true
+                testRunTitle: ${{ BuildPlatform.name }} sample build tests
+              displayName: ${{ BuildPlatform.name }} sample build tests
```

Comment:
I think Rui is saying these lines should be in the cake target and then have just this in the yaml:

```pwsh
- pwsh: ./build.ps1 --target=dotnet-samples --verbosity=diagnostic
   displayName: 'Test Samples'
```

I also think the templates and samples tests should be different projects so we don't need filters - and the test code can live next to areas.

Reply:
Sorry misunderstood that part, I added a new cake task to run these tests. As for individual .csproj files per test type, I would rather avoid that if possible. I think it is useful to have a generic test solution that can be used to build/launch any project, and serve as a place to add tests that don't have a specific src area in the future if needed.

Reply:
On second thought, running the test assembly directly with the dotnet yaml task takes a lot of effort out of test result xml creation and publishing, so I would prefer to keep that if possible.